numpy/lib/index_tricks.py

```diff
@@ -146,24 +146,25 @@ def __init__(self, sparse=False):
     def __getitem__(self, key):
         try:
             size = []
-            typ = int
+            # Mimick the behavior of `np.arange` and use a data type
+            # which is at least as large as `int` / `np.long`
```

Comment:
```suggestion
            # which is at least as large as `np.int_`
```
`np.long` is deprecated

Reply:
Done as of https://github.com/numpy/numpy/pull/16971/commits/e55436361436bf3cd024e54610d068974b89cc89.